ipv-core/src/main/java/com/ibm/whc/deid/providers/masking/BinningMaskingProvider.java

```diff
@@ -19,6 +19,12 @@ public class BinningMaskingProvider extends AbstractMaskingProvider {
   private final int startValue;
   private final boolean useStartValue;
   private final int normalizedStartValue;
+  private boolean useSingleBucketOverThreshold;
+  private double singleBucketOverThresholdValue;
+  private String singleBucketOverThresholdReplacement;
+  private boolean useSingleBucketUnderThreshold;
+  private double singleBucketUnderThresholdValue;
+  private String singleBucketUnderThresholdReplacement;
```

Comment:
I would mark all of these variables "final" like the ones above.  In Java, that means the value can't change after it is initially set.  That is the case here.  I believe it helps the compiler make some optimizations and it helps for making the class multi-threaded.